src/views/CoCurricularTranscript/Components/Experience/index.tsx

```diff
@@ -35,4 +43,19 @@ const formatDuration = ({ Job_Start_Date, Job_End_Date }: StudentEmployment) =>
   }
 };
 
+const newJobTitle = (
+  { Job_Department_Name, Job_Title }: StudentEmployment,
+  previousTitles: string[],
+  setPreviousTitles: Dispatch<SetStateAction<string[]>>,
+) => {
+  if (!previousTitles.includes(Job_Title)) {
+    previousTitles.push(Job_Title);
+    setPreviousTitles(previousTitles);
+    return Job_Department_Name === Job_Title.split(':')[0]
+      ? Job_Title
+      : `${Job_Department_Name}, ${Job_Title}`;
+  }
+  return '';
+};
+
```

Comment:
There is another way to do this, which in my opinion is both easier and more maintainable.

Basically, you can use [`Object.groupBy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy) to group the experiences by Job_Title, and display the title once per group of jobs with the same title.

The grouping is as simple as:
```typescript
const grouped_experiences = Object.groupBy(experiences, (experience) => experience.Job_Title);
```

Reply:
To give a more complete example, here's a simplified experiences array:
```javascript
const jobs = [
    {title: 'Dishwasher', start: new Date('2016-08-20'), end: new Date('2017-05-15')},
    {title: 'RA', start: new Date('2017-08-01'), end: new Date('2018-05-25')},
    {title: 'RA', start: new Date('2018-08-02'), end: new Date('2019-05-10')}
];
```

and we can group it like so:
```javascript
const jobsByTitle = Object.groupBy(jobs, job => job.title);
```

we can convert that object to an array of objects, where each object in the array has a `Job_Title` property which is the title of all the jobs in that group, and a `jobs` property which is the array containing all the jobs in that group:
```javascript
const jobGroups = Object.entries(jobsByTitle).map(([title, jobs]) => ({Job_Title: title, jobs}))
```

then we can render those job groups:
```jsx
const Experience = ({Job_Title, jobs}) => {
return (
    <div className={styles.experience_transcript_activities}>
        <div className={styles.organization_role}>{Job_Title}</div>
        {jobs.map((job) => (
           <div>{formatDuration(job)}</div>
        )}
    </div>
)
```

Reply:
Thank you for your suggestion. Can you explain why using group is easier and more maintainable? This code takes an array of information one by one from the ordered list. If I use group, it need an extra function to extract data from the group and a function that is similar to the upper code to display one title per group and to ignore job titles in the group, which is more complicate.

Reply:
Because your current code is using React's `useState` hook to keep track of whether a certain title has been seen yet or not, you have to update state (and thus re-render the entire transcript) once for each experience. If you instead group experiences by job title before passing the data to React (i.e. within `services/transcript.ts`), you will already have all the data calculated and won't have to re-render any extra times.

Reply:
Thank you for your answer! I tried to group experiences within `services/transcript.ts` (line 51-78), but the type of data will be changed from `groupedMembershipHistory` to array if I group by job title. Is there better way to group experience or improve the code without `groupBy`?